src/adapters/mpd.cpp

```diff
@@ -29,15 +30,26 @@ namespace mpd {
 
   void check_errors(mpd_connection* conn) {
     check_connection(conn);
+
+    string err_msg;
```

Comment:
I don't think we need to create `err_msg`, just call `throw foo_error(mpd_connection_get_error_message(conn), err)` like in the default case.

Reply:
I actually forgot to use `err_msg` in the default case. We need to temporarily store the message because `mpd_connection_clear_error` clears all error information.